### DISCUSSION TITLE
Best way to access user table if you don't have a separate server

### DISCUSSION THREAD

**Discussion by veggie2u** [supabase/supabase#11368] [Questions] [answered]
I have a solid-js front end and a supabase backend, google auth with supabase for login. My app is private. People apply to be accepted to the group, then they use google for login. They have to use the same google email to login as they provided when they applied. 

The problem is people being people, they try to log in with the wrong account, or forget what they signed up with, or whatever. So I am trying to create an admin screen so that an admin user can see the list of people that logged in with google and can figure out what went wrong, and delete users from that user table to remove a login.

I have no middleware/server to use my secret key from. Can I use it in edge functions? I only saw anon access in the examples. Is there another way?

**Comment by GaryAustin1** [comment, marked as answer]
You can use it in edge functions with service role.

You can also use security definer rpc calls IF you provide security in those Postgres functions so they are only used by your approved users.  Could be done by UUID or table look up with UUIDs that are acceptable to get results or use the function.

**Reply by veggie2u** [reply]
Hmmm. I will have to think about security definer rpc calls. I did make a db function (which fails due to security) but didn't try to use security definer and handle security inside the function. In the edge functions I see you get the ANON key like this: **Deno.env.get('SUPABASE_ANON_KEY')** do you know what the text is for the service key? SUPABASE_SERVICE_KEY maybe?

**Reply by GaryAustin1** [reply]
From the docs...
https://supabase.com/docs/guides/functions/quickstart#default-secrets

**Reply by veggie2u** [reply]
Thanks for that link. I missed that page.
